### PR TITLE
Wallet fix activity bug 

### PR DESCRIPTION
**Pull request by Jibz1**
- Fixed wallet activities feed 
- Show Capy images in the activities feed

Screenshot 2022-10-28 at 3 56 58 PM

### CODE REVIEW

File: apps/wallet/src/ui/app/components/transactions-card/index.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ function TransactionCard({ txn }: { txn: TxResultState }) {
     const transferType =
         txn.kind === 'Call' ? 'Call' : txn.isSender ? 'Sent' : 'Received';
 
-    const amount = txn?.balance || txn?.amount || txn?.txGas || 0;
+    const amount = txn?.amount || txn?.txGas || 0;
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/transactions-card/index.tsx:50]: Why this removal?

File: apps/wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 2]
```diff
@@ -108,71 +120,77 @@ export const getTransactionsByAddress = createAsyncThunk<
...
             });
 
         // Get all objectId and batch fetch objects for transactions with objectIds
         // remove duplicates
+
         const objectIDs = [
-            ...new Set(resp.map((itm) => itm.objectId).filter(notEmpty)),
+            ...new Set(
+                resp
+                    .filter(notEmpty)
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/redux/slices/txresults/index.ts:190]: Do we need the double notEmpty filter?

File: apps/wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 3]
```diff
@@ -69,16 +70,27 @@ const deduplicate = (results: string[] | undefined) =>
...
+    const events = txEffects?.events || [];
+    const objectIDs = events
+        ?.map((event: SuiEvent) => {
+            const data = Object.values(event).find(
+                (itm) => itm?.recipient?.AddressOwner === address
+            );
+            return data?.objectId;
+        })
+        .filter(notEmpty);
+    //
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/redux/slices/txresults/index.ts:90]: Remove empty comment

File: apps/wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 4]
```diff
@@ -69,16 +70,27 @@ const deduplicate = (results: string[] | undefined) =>
...
 const moveCallTxnName = (moveCallFunctionName?: string): string | null =>
     moveCallFunctionName ? moveCallFunctionName.replace(/_/g, ' ') : null;
 
+// Get objectId from a transaction effects -> events where recipient is the address
+const getTxnEffectsEventID = (
+    txEffects: TransactionEffects,
+    address: string
+): string[] => {
+    const events = txEffects?.events || [];
+    const objectIDs = events
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/redux/slices/txresults/index.ts:82]: Do we really want to go through every event group?

File: apps/wallet/src/ui/app/redux/slices/txresults/index.ts [hunk 5]
```diff
@@ -190,16 +208,21 @@ export const getTransactionsByAddress = createAsyncThunk<
...
+
             return {
                 ...itm,
+                coinType,
+                coinSymbol: coinType && Coin.getCoinSymbol(coinType),
                 ...(objectTxObj
                     ? {
-                          description: objectTxObj.data.fields.description,
-                          name: objectTxObj.data.fields.name,
+                          //Temporary solution to deal unknown object type
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/redux/slices/txresults/index.ts:219]: This isn't really temporary.

File: apps/wallet/src/ui/app/components/transactions-card/index.tsx (hunk 1 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/transactions-card/index.tsx:50]: `txn.balance` seems to refer to the balance of the sender.

File: apps/wallet/src/ui/app/redux/slices/txresults/index.ts (hunk 4 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/redux/slices/txresults/index.ts:82]: Do you mean stop the loop when we find the first matching item?

Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/redux/slices/txresults/index.ts:82]: I mean more is everything in this `events` object useful to look at?

**Review by Jordan-Mysten — APPROVED** [review]

**Pull request merged**